src/github.com/getlantern/lantern-mobile/test/lib/suite.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+
+import os
+import time
+import unittest
+from appium import webdriver
+# from time import sleep
+# from selenium.common.exceptions import WebDriverException
+
+
+def log(msg):
+    print (time.strftime("%H:%M:%S") + ": " + msg)
+
+
+class ParametrizedTestCase(unittest.TestCase):
+    """ TestCase classes that want to be parametrized should
+        inherit from this class.
+    """
+    def __init__(self, methodName='runTest', param=None):
+        super(ParametrizedTestCase, self).__init__(methodName)
+        self.param = param
+
+    @staticmethod
+    def parametrize(testcase_klass, param=None):
+        """ Create a suite containing all tests taken from the given
+            subclass, passing them the parameter 'param'.
+        """
+        testloader = unittest.TestLoader()
+        testnames = testloader.getTestCaseNames(testcase_klass)
+        suite = unittest.TestSuite()
+        for name in testnames:
+            suite.addTest(testcase_klass(name, param=param))
+        return suite
+
+
+class TestLantern(ParametrizedTestCase):
+    def setUp(self):
+        log("Starting Appium test using device '%s'" % self.device_name)
+        tm = time.strftime("%Y%m%dT%H%M%S", time.gmtime())
+        self.screenshotDir = os.path.join('./screenshots', tm + '_' + self.device_name)
+        log("Will save screenshots at: " + self.screenshotDir)
+        try:
+            os.makedirs(self.screenshotDir)
+        except OSError, e:
+            log("Fail to create screenshot directory %s: %s", self.screenshotDir, e)
+
+        log("WebDriver request initiated. Waiting for response, this typically takes 2-3 mins")
+        self.driver = webdriver.Remote(self.hub_url, self.desired_caps)
+        log("WebDriver response received")
+
+    def tearDown(self):
+        log("Quitting")
+        self.driver.quit()
+
+    def testStartAndQuit(self):
+        log("Test: start and quit app")
+        self.driver.save_screenshot(self.screenshotDir + "/1_appLaunch.png")
+
+        # self.by_id('settings_icon').click() # it doesn't work on API 17 or below!
```

Comment:
These tests look awesome, @fffw! Elements are defined in the XML files in the ``layout`` resources directory. It would be cool if we could add a test here to toggle the on/off switch (the ToggleButton is defined in the ``activity_lantern_main.xml`` file).